### ISSUE TITLE
Resilience4j always picks own default values

### ISSUE THREAD

**Issue by kencarvo** [resilience4j/resilience4j#1475]
Resilience4j version: 1.4.0

Java version:11

I have configured resilience4j defaults in application yaml file under a spring profile.
This is a Spring Boot 2 application. 
When I run a mockMvc test trying to use this configuration,, it is picking the default resilience4j settings. For eg. Default Resilience4j setting is COUNT_BASED and in my 
application yaml I have set the default settings to TIME_BASED.  Same issue for all other parameters.
In short it is not picking the configuration from the yaml file.
Please assist this urgently as I have lost few days over this issue but not able 
to figure out what is going astray.

**Comment by RobWin** [comment]
Hi,
I can't help without seeing your test or configuration

**Comment by kencarvo** [comment]
I can share only major pieces within the application.yaml file.

spring:
  application:
  name: fndkgn-v2

resilience4j.circuitbreaker:
  configs:
    default:
        registerHealthIndicator: true
...
....     slidingWindowType: TIME_BASED
....     minimumNumberOfCalls: 20
....     


Within the test case:

@ActiveProfiles("local")
@AutoConfigureWireMock....
@AutoConfigureMOckMvc
..
public class Mnjjkfsf {

    @AutoWired
     CircuitBreakerRegistry cbRegistry;


  voic getXXXCircuitBreakerTest ()  {

    CircuitBreaker circuitBreaker = cbRegistry.circuitBreaker(name: "Polly");
    CircuitBreakerConfig cbF = ......;

     System.out.println( cbfConfig.getMinimumNumberOfCalls ) ;  gives 100

    and not 20...
    
     and WindowType comes as COUNT_BASED instead of TIME_BASED.
  }

}

The version being used is 1.4.0

**Comment by kencarvo** [comment]
Hello Robein... Keenly awaiting your reply.. Any suggestion is welcome

**Comment by RobWin** [comment]
Unfortunately, I don't know.
The question is too Spring Boot-specific. I don't know how `@ActiveProfiles` actually works. Not sure if your `application.yaml` is read when you only specify "@ActiveProfiles("local")". 
I guess you need to debug the `ApplicationContext`.

**Comment by kencarvo** [comment]
@activeprofiles is telling the test to consider only these profiles as active among the many that have been defined. 
How can I debug the application context or know whether the application yaml is being read. Thanks for the answer cause I realize that I had assumed many things like the yaml being read by default etc.

**Comment by RobWin** [comment]
You can use any IDE debugger and debug into whatever Spring Boot class composes the application config files. 
Resilience4j is only using `ConfigurationProperties` annotations.

**Comment by kencarvo** [comment]
Thanks I will work on that solution.
